cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -0,0 +1,258 @@
+        // This list needs to match the IDBMMS list of baseline services
+        // https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/
+        //   services/idbrokermappingmanagement/src/main/java/com/cloudera/thunderhead/
+        //   service/idbrokermappingmanagement/server/IdBrokerMappingManagementService.java#L113
+        List<String> rangerAuditUsers = Arrays.asList("kafka", "solr", "knox", "atlas");
+
+        AccountMappingBase accountMappings = cloudFileSystem.getAccountMapping();
+        if (accountMappings != null) {
+            Set<String> roleArns = getRoleArnsForUsers(rangerAuditUsers, accountMappings.getUserMappings());
+
+            List<String> policyFileNames = Arrays.asList(
+                "aws-cdp-bucket-access-policy.json",
+                "aws-cdp-ranger-audit-s3-policy.json"
+            );
+            String s3GuardDynamoDBTableName = cloudFileSystem.getS3GuardDynamoTableName();
+            if (s3GuardDynamoDBTableName != null) {
+                policyFileNames.add("aws-cdp-dynamodb-policy.json");
+            } else {
+                s3GuardDynamoDBTableName = "";
+            }
+
+            for (StorageLocationBase location : cloudFileSystem.getLocations()) {
+                if (CloudStorageCdpService.RANGER_AUDIT.equals(location.getType())) {
+                    String storageLocationBase = location.getValue()
+                                                    .replace("s3a://", "")
+                                                    .replace("/ranger/audit", "");
+                    String datalakeBucket = storageLocationBase.split("/", 2)[0];
+                    checkRolePermissions(roleArns, policyFileNames, datalakeBucket,
+                        storageLocationBase, s3GuardDynamoDBTableName);
+                }
+            }
+        }
+    }
+
+    /**
+     * Retrieve the unique set of role arns for the list of provided users
+     * @param users list of users to filter by
+     * @param userMappings user mappings to filter on
+     * @return set of role arns
+     */
+    private Set<String> getRoleArnsForUsers(List<String> users, Map<String, String> userMappings) {
+        return userMappings.entrySet().stream()
+                    .filter(m -> users.contains(m.getKey()))
+                    .map(Map.Entry::getValue).collect(Collectors.toUnmodifiableSet());
+    }
+
+    private void checkRolePermissions(Collection<String> roleArns, List<String> policyFileNames,
+                                        String datalakeBucket, String storageLocationBase,
+                                        String s3GuardDynamoDBTableName) {
+        Set<String> failedActions = new HashSet<>();
+        for (String policyFileName : policyFileNames) {
+            try {
+                String policyJsonTemplate = getResourceFileAsString("definitions/" + policyFileName);
+                if (policyJsonTemplate != null) {
+                    String policyJson = policyJsonTemplate
```

Comment:
We already using handlebar in the project which is a templating engine. Can we use that here ?

Reply:
I looked into this. `handlebars` is only currently used in the `template-manager-core` module. It seems to be pulled in for building template objects like blueprints.

I don't really like the string replacement, but I wanted to keep the policy json files human readable. This should allow @dbialek2 to reference/embed them in the documentation (https://docs.cloudera.com/management-console/cloud/environments/topics/mc-idbroker-minimum-setup.html) instead of docs having to keep a separate copy of the policy json files.

ie: https://raw.githubusercontent.com/risdenk/cloudbreak/CB-3513/cloud-aws/src/main/resources/definitions/aws-cdp-bucket-access-policy.json (eventually will be in CB repo instead of in my fork but same idea)

Reply:
👍